pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -364,6 +388,11 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 			Message: reconcileCompletedMessage,
 		})
 
+		if r.upgradeMode { // update the new image only when upgrade is completed
+			instance.Status.UpdateVersion(hcoVersionName, r.ownVersion)
+			r.upgradeMode = false
```

Comment:
I think we should also log something with reqLogger.Info(...)  explicitly reporting that the upgrade has been completed